frontend/apps/filemanager/filemanagersetdefaults.lua

```diff
@@ -356,4 +356,28 @@ function SetDefaults:saveSettings()
     self.settings_changed = false
 end
 
+function SetDefaults:saveBeforeExit(callback)
+    if self.settings_changed then
+        UIManager:show(ConfirmBox:new{
+            text = _("You have unsaved default settings. Save them now?\nTap \"Cancel\" to return to KOReader."),
+            ok_text = _("Save"),
+            ok_callback = function()
+              self.settings_changed = false
+              self:saveSettings()
+              callback()
```

Comment:
Does it make sense to check for callback ~= nil, so that the save dialog can be triggered without further action?
The "Don't save" action would become a "Don't save now", the changes would still be included in a next save.

Reply:
When you said the real function as a callback I thought you simply meant something like `UIManager:close(menu_container)`.

Reply:
Not sure I understand what you mean.
If you mean call `UIManager:close(<some_parameter>)` instead of `callback()`, I feel uncomfortable doing that because `ReaderMenu:exitOrRestart()` has quite a bit more logic in it than `FileManagerMenu:exitOrRestart()`, so I'd rather preserve the code.

Reply:
And moreover, I wrote most of that logic for good reason! ;-)